design/secrets_provider_milestone1_design.md

```diff
@@ -42,54 +42,103 @@
+For Milestone 3, running by the app as **Init Container** or **Sidecar** allows writing and updating the secrets to files in a shared volume with the app.
 
-Additionally, when using Deployment to spin up a pod and the task finishes, the will container exit and be deleted. Now that the Secrets Provider needs to sit separately, when that container exists so too will the pod. Therefore, when using Deployment, Kuberentes will try to restart the Pod. After a few attempted restarts, the pod will error in a BackOff state. Because of this, we are unable to use Deployment to spin up a pod and must use a Job.
+###### What benefits does this solution provide?
+
+1. **Clarity**
+   Allowing customer to select the deployment to get the desired value
+2. **Maintainability**
+   Using one image for all variations simplifies code maintenance 
+3. **Reusability**
+   Using one image for all variations allows reuse of existing code across all variations
+4. **Extensibility**
+   Supporting various flows using the same solution opens the door for more secrets sources and targets
+
+###### What drawbacks does this solution have?
+
+**Job** is not an existing application identity granularity.
+As a result, one can define the host representing the Secrets Provider in Conjur policy using only **namepsace** and/or **service account** granularities.
+
+###### How the drawbacks can be handled?
+
+To handle the missing **Job** granularity there are 2 options:
+
+|      | Solution                                                     | Pros                                                         | Cons                                                         | Cost Estimation |
+| ---- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | --------------- |
+| 1    | Keep using only **namepsace** and/or **service account** granularities. | - Since Secrets Provider runs as a separate entity, we suggest  creating service account dedicated for Secrets Provider.<br />Then, **service account** granularity serves exact match and **Job** granularity is redundant<br />- No code changes | - Service account might be reused, which allows other apps to authenticate using Secrets Provider host | Free            |
+| 2    | Run as non-terminating  Pod                                  | - Allows specific identification of the Secrets Provider **Job**, which enhances security | - Costly; requires code changes, verify compatibility, tests, docs<br />- Redundant if **service account** serves only the Secrets Provider | 10 days         |
```

Comment:
I don't agree that this will take 10 days.
`Run as non-terminating  Pod` it is not possible to run as a non-terminating Pod ...  because of my explaining 
```
Additionally, when using Deployment to spin up a pod and the task finishes, the will container exit and be deleted. Now that the Secrets Provider needs to sit separately, when that container exists so too will the pod. Therefore, when using Deployment, Kuberentes will try to restart the Pod. After a few attempted restarts, the pod will error in a BackOff state. Because of this, we are unable to use Deployment to spin up a pod and must use a Job.
```